client/src/components/DataFiles/DataFilesBreadcrumbs/DataFilesBreadcrumbs.jsx

```diff
@@ -155,52 +199,89 @@ const DataFilesBreadcrumbs = ({
       }
     }, '');
 
+  const lastTwoPaths = paths.slice(-1);
+  const lastTwoPathComps = pathComps.slice(-1);
+  const reversedPath = paths.reverse();
+
   return (
-    <div className={`breadcrumbs ${className}`}>
-      {scheme === 'projects' && (
-        <>
-          <RootProjectsLink
-            api={api}
-            section={section}
-            operation={operation}
-            label="Shared Workspaces"
-          />{' '}
-          {system && `/ `}
-        </>
+    <div className="breadcrumb-container">
+      <div id="path-button-wrapper">
+        <ButtonDropdown
+          isOpen={dropdownOpen}
+          toggle={toggleDropdown}
+          id="go-to-button-dropdown"
+          className="go-to-button-dropdown"
+        >
+          <DropdownToggle tag={Button}>Go to ...</DropdownToggle>
+          <DropdownMenu>
+            {reversedPath.slice(1, reversedPath.length).map((path, index) => {
+              const folderName = path.split('/').pop();
+              return (
+                <>
+                  <Link key={index} to={handleNavigation(path)}>
+                    <DropdownItem>
+                      <i className="icon-folder" />
+                      {folderName.length > 20
+                        ? folderName.substring(0, 20)
+                        : folderName}
+                    </DropdownItem>
+                  </Link>
+                </>
+              );
+            })}
+            <DropdownItem divider />
+            <Link to={handleNavigation(homeDir)}>
+              <DropdownItem>
+                <i className="icon-my-data" />
+                <span className="multiline-menu-item-wrapper">
+                  {systemName || 'Shared Workspaces'}
+                  <small>Root</small>
+                </span>
+              </DropdownItem>
+            </Link>
+          </DropdownMenu>
+        </ButtonDropdown>
+      </div>
+      <div className={`breadcrumbs ${className}`}>
+        {lastTwoPathComps.length === 0 ? (
+          <span className="system-name truncate">
+            {systemName || 'Shared Workspaces'}
+          </span>
+        ) : (
+          lastTwoPathComps.map((pathComp, i) => {
+            if (i === lastTwoPaths.length - 1) {
+              return (
+                <span key={uuidv4()} className="truncate">
+                  {pathComp}
+                </span>
+              );
+            }
+            return (
+              <React.Fragment key={uuidv4()}>
+                <span className="vertical-align-separator">/</span>
+                <BreadcrumbLink
+                  api={api}
+                  scheme={scheme}
+                  system={system}
+                  path={lastTwoPaths[i]}
+                  section={section}
+                >
+                  <>{pathComp}</>
```

Comment:
Do we want to truncate-middle the paths, rather than truncate-end?
![Screenshot 2023-10-04 at 7 10 03 PM](https://github.com/TACC/Core-Portal/assets/20326896/68a1aef7-fc16-433c-bd7b-2348d4aac1d6)

Reply:
<img width="1423" alt="truncate-middle" src="https://github.com/TACC/Core-Portal/assets/50084480/e6918f8b-9897-4efc-8d15-ace2cdfeda9c">
Truncate middle example for long current directory names

Reply:
Truncate Middle functionality is now added per Sal's suggestion to used truncateMiddle utility